src/backend/core/authentication.py

```diff
@@ -1,59 +1,107 @@
 """Authentication for the People core app."""
-from django.conf import settings
-from django.utils.functional import SimpleLazyObject
-from django.utils.module_loading import import_string
+
+from django.core.exceptions import SuspiciousOperation
+from django.db import models
 from django.utils.translation import gettext_lazy as _
 
-from drf_spectacular.authentication import SessionScheme, TokenScheme
-from drf_spectacular.plumbing import build_bearer_security_scheme_object
-from rest_framework import authentication
-from rest_framework_simplejwt.authentication import JWTAuthentication
+import requests
+from mozilla_django_oidc.auth import (
+    OIDCAuthenticationBackend as MozillaOIDCAuthenticationBackend,
+)
+
+from .models import Identity
+
 
+class OIDCAuthenticationBackend(MozillaOIDCAuthenticationBackend):
```

Comment:
Since you are mentioning customizations for AgentConnect, should we name it AgenceConnectOIDCAuthenticationBackend? Or is it in fact generic and we can remove the mention to AgentConnect in the docstring?